packages/node-resolve/test/test.js

```diff
@@ -257,6 +257,36 @@ test('allows custom moduleDirectories with legacy customResolveOptions.moduleDir
   t.snapshot(warnings);
 });
 
+test('custom moduleDirectories do not support nested dependencies', async (t) => {
```

Comment:
I think it's good to test this but it would be nice to also actually flag this as a proper error if the user tries to provide a path instead of name for this option.

Please could we update the code to something like:
```js
if (moduleDirectories.some((name) => name.includes('/'))) {
  throw new Error(
    '`moduleDirectories` option must only contain directory names. If you want to load modules from a custom directory see `modulePaths`.'
  );
}
```
in index.js

Then this test can just check that
```js
nodeResolve({
  moduleDirectories: ['some/path']
})
```
throws that error

Reply:
Yeah, that sounds awesome!  Would this change be breaking though?  Should it require a major version bump?

Reply:
If you use a path doesn’t it just get ignored currently?

If that’s the case and it silently fails (with the warning when module not found) IMO this doesn’t need to be a breaking change as it’s highlighting an invalid config

Reply:
What’s the existing behaviour you’re relying on?

Reply:
If you pass a relative path, it appears that the same logic for folder names is applied.  i.e.  When searching for node modules, rollup will recursively walk up the filesystem, at each level looking for a the full subpath specified.

I think we inadvertently assumed that moduleDirectories would allow us to specify a NODE_PATH, and for a little while we were accidentally getting away with it.

Reply:
Interesting.  This readme says it's meant to just contain directory names, and the doc for the `resolve` package we use says

>opts.moduleDirectory - directory (or directories) in which to recursively look for modules. default: "node_modules"

and [the tests for it are only for names](https://github.com/browserify/resolve/blob/dc8ad1945c663ba47821a269aefb5549f6177cf7/test/module_dir.js).

Given that, I think the fact that it did kind of work with a path was unintentional/undocumented, and therefore IMO erroring would be better and this would be fine as a patch. Especially given if people were misusing `moduleDirectories` when they see the error there will be the correct option available for them to to move too.

@shellscape might have a different opinion though :)

Reply:
sounds reasonable to me

Reply:
I have rebased and pushed a change addressing this feedback.  Let me know what you think!

Reply:
Sorry missed your reply